panic on nil dereference in debugging.ContainerManager

I've encountered the following panic a couple of times with `skaffold debug` when trying to dig into #5946.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x103b40054]

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/debugging.(*ContainerManager).Stop(0x14000d7a390)
	/Users/bdealwis/Projects/Skaffold/repo-skaffold/pkg/skaffold/kubernetes/debugging/container_manager.go:88 +0x34
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/v1.(*SkaffoldRunner).Dev(0x140004a3500, 0x1045da458, 0x140004ba0c0, 0x1045939e8, 0x1400041ac40, 0x14000551f98, 0x1, 0x1, 0x10458c808, 0x140004791a0)
	/Users/bdealwis/Projects/Skaffold/repo-skaffold/pkg/skaffold/runner/v1/dev.go:333 +0x10c0
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.runDev.func5(0x1045fc1d8, 0x140004a3500, 0x14000551cd0, 0x1, 0x1, 0x0, 0x0)
	/Users/bdealwis/Projects/Skaffold/repo-skaffold/cmd/skaffold/app/cmd/dev.go:68 +0x1bc
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.withRunner(0x1045da458, 0x140004ba0c0, 0x1045939e8, 0x1400041ac40, 0x14000873b00, 0x105c7ca68, 0x8)
	/Users/bdealwis/Projects/Skaffold/repo-skaffold/cmd/skaffold/app/cmd/runner.go:55 +0xd0
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.runDev(0x1045da458, 0x140004ba0c0, 0x1045939e8, 0x1400041ac40, 0x0, 0x0)
[...]
```

### Information

- Skaffold version: HEAD and 1.26.1
- Operating system: macOS

--- a/pkg/skaffold/kubernetes/debugging/container_manager.go
+++ b/pkg/skaffold/kubernetes/debugging/container_manager.go
@@ -45,9 +45,10 @@ func NewContainerManager(podSelector kubernetes.PodSelector) *ContainerManager {
 	// Create the channel here as Stop() may be called before Start() when a build fails, thus
 	// avoiding the possibility of closing a nil channel. Channels are cheap.
 	return &ContainerManager{
-		podWatcher: kubernetes.NewPodWatcher(podSelector),
-		active:     map[string]string{},
-		events:     make(chan kubernetes.PodEvent),
+		podWatcher:  kubernetes.NewPodWatcher(podSelector),
+		active:      map[string]string{},
+		events:      make(chan kubernetes.PodEvent),
+		stopWatcher: func() {},
 	}
 }
 
@@ -65,6 +66,8 @@ func (d *ContainerManager) Start(ctx context.Context, namespaces []string) error
 	d.stopWatcher = stopWatcher
 
 	go func() {
+		defer stopWatcher()
+
 		for {
 			select {
 			case <-ctx.Done():
@@ -84,9 +87,10 @@ func (d *ContainerManager) Start(ctx context.Context, namespaces []string) error
 
 func (d *ContainerManager) Stop() {
 	// if nil then debug mode probably not enabled
-	if d != nil {
-		d.stopWatcher()
+	if d == nil {
+		return
 	}
+	d.stopWatcher()
 }
 
 func (d *ContainerManager) Name() string {

